### PR TITLE
[CDAP-1245] return 200 with empty list for streams and datasets that are not used by a flow

### PR DESCRIPTION
**Pull request by anew**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java [hunk 1]
```diff
@@ -1230,8 +1230,12 @@ private void programListByDataAccess(HttpRequest request, HttpResponder responde
     }
   }
 
+  /**
+   * @return list of program records; or null if the stream or dataset does not exist.
```
Comment (awholegunch) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java]: @return list of program records or null if the stream or dataset do not exist
\[no closing period on parameters and return descriptions]

Comment (awholegunch) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java]: though do you need to indicate that the list is empty if no programs were found yet the data exists?

**Comment by gokulavasan** [comment]
@anew Can you add the JIRA associated with this, if any?

**Comment by anew** [comment]
It's the name of the branch CDAP-1245

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**